pkg/apis/externaldns/types.go

```diff
@@ -321,6 +325,8 @@ func (cfg *Config) ParseFlags(args []string) error {
 	app.Flag("publish-internal-services", "Allow external-dns to publish DNS records for ClusterIP services (optional)").BoolVar(&cfg.PublishInternal)
 	app.Flag("publish-host-ip", "Allow external-dns to publish host-ip for headless services (optional)").BoolVar(&cfg.PublishHostIP)
 	app.Flag("always-publish-not-ready-addresses", "Always publish also not ready addresses for headless services (optional)").BoolVar(&cfg.AlwaysPublishNotReadyAddresses)
+	app.Flag("node-port-enabled", "Enable the creation of an SRV record for NodePorts alongside the other DNS records (optional)").BoolVar(&cfg.NodePortEnabled)
+	app.Flag("node-port-selector", "The selector to filter nodes by when selecting NodePort targets (defaults to 'node-role.kubernetes.io/node=', the k8s default for all worker nodes)").Default(defaultConfig.NodePortSelector).StringVar(&cfg.NodePortSelector)
```

Comment:
I am not sure about those options. 

1. `node-port-enabled` naming is a bit confusing (`NodePort` vs `NodePort port`). Now it implies a bit that the `NodePort` can be enabled/disabled here, colliding with `service-type-filter`.

2. `node-port-selector` should be disabled by default, it is breaking change for clusters that are not using the default `node-role.kubernetes.io/node=`, very likely not compatible with AWS EKS clusters running on Fargate.